rosbag2_transport/test/rosbag2_transport/test_record.cpp

```diff
@@ -151,9 +151,11 @@ TEST_F(RecordIntegrationTestFixture, receives_latched_messages)
   // Publish messages before starting recording
   test_msgs::msg::Strings msg;
   msg.string_value = "Hello";
+  rclcpp::executors::SingleThreadedExecutor exec;
+  exec.add_node(publisher_node);
```

Comment:
This is still throwing an exception in tests; in particular, the tests show:

```
unknown file: Failure
C++ exception with description "Node has already been added to an executor." thrown in the test body.
```

I think that is because we are adding the publisher_node to an executor here, but then down below `rosbag2_test_common::spin_and_wait_for` is also creating it's own executor and adding the node.

I think you can probably fix this by doing `exec.remove_node()` right before `rosbag2_test_common::spin_and_wait_for`.